demo.py

```diff
@@ -0,0 +1,146 @@
+class Resource(models.Model):
+    client = ForeignKey(Client)
+    owned_by = ForgeignKey(Resource, null=True)
+    name = CharField(blank=False, max_length=128)
+    configuration = JSONField(null=False)
+    types = ResourceTypeRegistry()
```

Comment:
The resource type registry was to be a class (static) property that can enumerate all of the subclasses of `Resource`. But looking at it now, it's probably not necessary.

There is an example of its use as I was thinking below in this PR:

```
   mysql_server=Option(
                type=Resource.types.MysqlServer,
                null=True,
                default=None,
                description="Leave blank to use an ephemeral MySQl server on the sandbox VM.",
            ),
```

Although if we refactor to Django multi-table inheritance models instead of my idiosyncratic `configuration` scheme, it could probably look instead like:

```
class OpenEdxSandboxInstance(Resource):
    mysql_server = ResourceField(
        type=MysqlServer,
        null=True,
        default=None,
        help_text=_("Leave blank to use an ephemeral MySQl server on the sandbox VM."),
    )    
```

Where `ResourceField` is a subclass of `ForeignKey` that implements an appropriate `limit_choices_to` filter set so that each resource can only link to other resources in the same project, etc.